Feature/ui3

ThomasJGabrielsen opened the pull request:
Replace Main menu UI with more modern alternative, and expand on the menu scene with new models: 
![Screenshot 2024-12-11 205221](https://github.com/user-attachments/assets/41b276ab-fee0-4d4d-8e3f-577457a48eef)

ThomasJGabrielsen committed:
Improved button visual design

ThomasJGabrielsen committed:
Added physical signboard with logo

ThomasJGabrielsen committed:
reworked menu

ThomasJGabrielsen committed:
Basic functionality fix

ThomasJGabrielsen committed:
Reintroduced props to menu.

Adjusted scale and positioning of UI elements
Forced a vector to be a value in case random returns (0,0) in TilingCowboys.cs
Commented out camera moving to options in MainMenuMoveCamera.cs
Added comments to TilingCowboys.cs

ThomasJGabrielsen committed:
Re-enabled correct camera

toberge commented on Assets/Scripts/UI/MainMenu/MainMenuMoveCamera.cs:
```diff
@@ -29,8 +31,8 @@ private void Start()
-        LeanTween.sequence()
-            .append(LeanTween.rotateY(mainMenuCamera.gameObject, 10, cameraSpeed).setEaseInOutQuart());
+        //LeanTween.sequence()
+        //    .append(LeanTween.rotateY(mainMenuCamera.gameObject, 10, cameraSpeed).setEaseInOutQuart());
```
muh settings view -_-

ThomasJGabrielsen replied:
Goodbye

toberge replied:
...I'd like the settings view to be back though

Fueredoriku replied:
Fixed

toberge reviewed (changes requested):
Overall very nice, especially
- cool new buttons that have better transitions (could perhaps use a background in some cases but ehhh)
- nice that we also have the sliding cowbot backdrop for the descriptions
- torii gate

However, I found a few issues:

- keybind and look options still have black text
- keybinding reset button looks a little scuffed
  ![image](https://github.com/user-attachments/assets/ba1209cf-c133-443e-ac76-2928a92b766d)
- cowbot background no longer moves on the loading screen :/
- queue meny looks weird w/o steam connection, and keyboard navigation breaks when you press down from the back button:
  ![image](https://github.com/user-attachments/assets/afb91ed6-fdc3-4faa-a4e5-9b606cba7d55)
- Pressing down from Three Strikes jumps straight to the back button in local play and friends only menu:
  ![image](https://github.com/user-attachments/assets/ea0a9e2b-6a22-4610-a557-a353ba542fd1)
- In the queue menu, it doesn't even move from that button:
  ![image](https://github.com/user-attachments/assets/498cbfc7-665c-4b40-bc87-d9f6d1d70d06)
- Custom gamemode menu doesn't have the fancy new background *but* white text:
  ![image](https://github.com/user-attachments/assets/032d21b5-7597-483c-a7bb-23a435ee8262)
- Pause menu has different text size for the two buttons
- and sometimes I need to press the movement key twice to actually move from button to button, so something seems to be mildly borked there?

toberge commented on Assets/Scripts/UI/TilingCowboys.cs:
```diff
@@ -0,0 +1,48 @@
+
+
+
+    // Start is called before the first frame update
```
pls remove standard comments here :))))

ThomasJGabrielsen replied:
Done

toberge commented on Assets/Scripts/UI/TilingCowboys.cs:
```diff
@@ -0,0 +1,48 @@
+
+public class TilingCowboys : MonoBehaviour
+{
+    [SerializeField] private RawImage cowboys;
```
typo, should be cowbots >:)

ThomasJGabrielsen replied:
Fixed

toberge commented on Assets/Materials/ThePit/metalTexture red.mat:
This material is used for boxes and platforms in The Pit, please create a new material instead.

![image](https://github.com/user-attachments/assets/317cfacf-3f7e-48de-9a93-80685a0bfd1e)

toberge replied:
Still an issue

Fueredoriku replied:
Fixed

Fueredoriku commented on Assets/TextMesh Pro/Resources/Fonts & Materials/LiberationSans SDF - Fallback.asset:
This one can be reverted

Fueredoriku commented on Assets/Textures/UI/Gradient.png:
Not sure where this is used, but I think there are gradient options on most things that can apply an effect like this without having to rely on a texture

toberge replied:
This is the background texture for the new buttons I think, so not really an issue there?

ThomasJGabrielsen replied:
This was an experiment to test different ways of highlighting the menu, we determined that the lack of gradient was best, so I'm removing the file next push.

ThomasJGabrielsen commented:
In regards to @toberge 's issue with the buttons being white when steam is not initialized, there seems to be a check somewhere in the code that disables the button component instead of unchecking the "interactable" boolean. I'm not sure where this check is and can't seem to find it. If you could change that @Fueredoriku as you are more familiar with that code, that would be great!

ThomasJGabrielsen committed:
Fixes based off of feedback:

- Renamed all instances of "cowboys" to "cowbots"
- Added TilingCowbots.cs to loading screen
- Fixed text color in keybind options submenu and custom game mode menu
- Added cowbots to custom game mode menu
- Adjusted text size in pause menu
- Added autosizing to certain numbered text elements to improve readability.
- Removed Gradient.png

Fueredoriku commented:
I'll take a gander! 👍

toberge commented on Assets/Materials/ThePit/metalTexture red.mat:
again, this entire file needs to be reverted

Fueredoriku replied:
Fixed

toberge commented:
![image](https://github.com/user-attachments/assets/2a29660d-ebcc-41e4-af3b-7261530c1a9b)

:o

toberge commented on Assets/Prefabs/Gamestate/MatchManager.prefab:
pause menu appears when entering match, might be related to a change here

Fueredoriku replied:
Fixed

toberge commented on Assets/Scripts/UI/TilingCowbots.cs:
```diff
@@ -0,0 +1,42 @@
+
+    void Start()
+    {
+        if (rotateCowbots)
```
Please make this the default for the loading screen at least.

Fueredoriku replied:
Fixed

toberge commented on Assets/Scripts/UI/TilingCowbots.cs:
```diff
@@ -0,0 +1,42 @@
+        }
+        else {
+            // Create a random angle to move towards
+            Vector2 angle = new Vector2(Random.Range(-1, 1), Random.Range(-1, 1));
```
The random.range calls here return integers since the parameters are integers. Try `Random.Range(-1f, 1f)` for proper rotation.

toberge commented:
overall there are a bunch of things that are fixed and a bunch of things that aren't. we could slip this in after a few more *important* fixes and make some changes afterwards

Fueredoriku commented:
Fixed the revert stuff + merge conflicts + the online buttons having the wrong background color 😁👍

toberge reviewed (approved):
it good now just pls fix camera angle for queue menu

Fueredoriku committed:
Fix online only buttons having wrong background

Fueredoriku reviewed (approved):
Looks really nice,
Lets gooo! 👍👍

toberge commented:
oh by the way please do a squash merge here (lots of the commits are like "fix problem X that appeared while working on this PR" and thus not necessary to keep)

ThomasJGabrielsen committed:
Updated pause menu to better fit new menu style
